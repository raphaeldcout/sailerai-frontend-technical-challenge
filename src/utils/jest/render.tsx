import { FC, ReactElement, ReactNode } from 'react';

import { ThemeProviderSailer } from '@/styles/withThemeProvider';
import { render, RenderOptions } from '@testing-library/react';

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <ThemeProviderSailer>{children}</ThemeProviderSailer>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };

