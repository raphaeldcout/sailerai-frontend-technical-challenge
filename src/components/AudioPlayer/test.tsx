import '@testing-library/jest-dom';

import { render } from '@/utils/jest/render';

import { AudioPlayer } from '.';

describe('AudioPlayer', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <AudioPlayer src="https://cdn.pixabay.com/audio/2021/08/09/audio_2f331550f9.mp3" />
    );

    expect(container).toMatchSnapshot();
  });
});
