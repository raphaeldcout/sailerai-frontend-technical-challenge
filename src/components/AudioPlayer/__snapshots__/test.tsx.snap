// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer should match snapshot 1`] = `
<div>
  <div
    class="sc-478098d3-0 NIKnl"
  >
    <audio
      class="sc-478098d3-1"
      controls=""
      src="https://cdn.pixabay.com/audio/2021/08/09/audio_2f331550f9.mp3"
    >
      <p
        class="sc-dc096805-0 lbvuzb"
        color="surface"
        variant="medium"
        weight="regular"
      >
        Seu navegador não suporta o elemento de áudio.
      </p>
    </audio>
  </div>
</div>
`;
